### PR TITLE
Expose computed field extents in generated code

### LINKED ISSUE #710: Expose the computed halo to be consumed by the frontend

### PR DESCRIPTION
**Pull request by havogt**
## Technical Description

Expose computed field extents in generated code as tuple-like objects. Additional named functions for accessing the elements could be added.

### Resolves / Enhances
Fixes #710.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: dawn/src/dawn/CodeGen/CodeGen.cpp [hunk 1]
```diff
@@ -444,5 +445,42 @@ void CodeGen::generateGlobalIndices(const iir::Stencil& stencil, Structure& sten
   }
 }
 
+namespace {
+std::string extent_to_string(iir::Extents const& extents, ast::GridType const& gridType) {
```
Comment (twicki) [inline dawn/src/dawn/CodeGen/CodeGen.cpp]:
```suggestion
std::string extentToString(iir::Extents const& extents, ast::GridType const& gridType) {
```

File: dawn/src/dawn/CodeGen/CodeGen.cpp [hunk 2]
```diff
@@ -444,5 +445,42 @@ void CodeGen::generateGlobalIndices(const iir::Stencil& stencil, Structure& sten
   }
 }
 
+namespace {
```
Comment (twicki) [inline dawn/src/dawn/CodeGen/CodeGen.cpp]: https://github.com/MeteoSwiss-APN/dawn/blob/4acd78b812d3d2303a9ac5308bc2ee957013d173/dawn/src/dawn/IIR/Extents.cpp#L371-L389

exists. Why do we need another one?

Comment (havogt) [reply dawn/src/dawn/CodeGen/CodeGen.cpp]: I first extended this to also capture my use-case (by adding parameters for which kind of bracket to use etc. But then there is another difference: the generic one doesn't know the GridType and therefore cannot properly cast a no-extent. So in the end I decided to go with the specialized one.

Comment (havogt) [reply dawn/src/dawn/CodeGen/CodeGen.cpp]: The other one is actually only in error messages. Not sure if we want to keep it.

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**